osu.Game/Collections/BeatmapCollectionManager.cs

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using osu.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Logging;
+using osu.Framework.Platform;
+using osu.Game.Beatmaps;
+using osu.Game.IO.Legacy;
+using osu.Game.Overlays.Notifications;
+
+namespace osu.Game.Collections
+{
+    public class BeatmapCollectionManager : CompositeDrawable
```

Comment:
rename to `CollectionManager` maybe? not sure we will ever have another type of collection (and the variable in `OsuGameBase` is already referring to it as `CollectionManager`).